server/src/main/java/com/cloud/user/AccountManagerImpl.java

```diff
@@ -1897,7 +1897,17 @@ public AccountVO updateAccount(UpdateAccountCmd cmd) {
                         "in the domain '" + domainId + "'.");
             }
 
+            Role role = roleService.findRole(roleId);
+            Long currentAccRoleId = account.getRoleId();
+            Role currentRole = roleService.findRole(currentAccRoleId);
+
+            if (role.getRoleType().ordinal() < currentRole.getRoleType().ordinal() && ((account.getType() == Account.Type.NORMAL && role.getRoleType().getAccountType().ordinal() > Account.Type.NORMAL.ordinal()) ||
+                    account.getType().ordinal() > Account.Type.NORMAL.ordinal() && role.getRoleType().getAccountType().ordinal() < account.getType().ordinal() && role.getRoleType().getAccountType().ordinal() > 0)) {
```

Comment:
This condition is hard to read,
Can you extract it to a method like `isRoleEscalation(...)` and indent according to the logic of it, please?
It would make sense to extract the entire block as well, i.e. `checkForRoleEscalation(...)`.